src/OrbitBase/include/OrbitBase/TaskGroup.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TASK_GROUP_H_
+#define ORBIT_BASE_TASK_GROUP_H_
+
+#include "OrbitBase/ThreadPool.h"
+
+namespace orbit_base {
+
+// Simple class used to parallelize a group of tasks. Add tasks to be run in parallel then call
+// "Wait()" to wait for all tasks to complete. Note that "Wait()" is called from the destructor if
+// it wasn't explicitly called. This class is not thread-safe. A TaskGroup object should be owned
+// and accessed by a single thread.
+//
+// Usage:
+//
+// void ProcessObjectsInParallel(std::vector<Object>& objects) {
+//   TaskGroup task_group(thread_pool_);
+//   for (Object& object : objects) {
+//     task_group.AddTask([&object]() { ProcessObject(object); });
+//   }
+// }
+//
+class TaskGroup {
+ public:
+  explicit TaskGroup(orbit_base::ThreadPool* thread_pool) : thread_pool_(thread_pool) {}
```

Comment:
Instead of taking a `ThreadPool`, could you take a pointer to `orbit_base::Executor`? ThreadPool implements the Executor interface which means this class could be used with other executors as well. That's also useful for testing.

Reply:
Sure thing!